### PR TITLE
SmoothMesh

### PR DESCRIPTION
**Pull request by friedmud**
Add a MeshModifier that smooths a mesh (respecting block boundaries too).

This requires a libMesh update to respect the block boundaries.

#11861

### CODE REVIEW

File: framework/src/meshmodifiers/SmoothMesh.C [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+void
+SmoothMesh::modify()
+{
+  // Check that we have access to the mesh
+  if (!_mesh_ptr)
+    mooseError("_mesh_ptr must be initialized before calling SmoothMesh::modify()");
+
+  LaplaceMeshSmoother lms(static_cast<UnstructuredMesh &>(_mesh_ptr->getMesh()));
+
+  lms.smooth(5);
```
Comment (permcody) [inline framework/src/meshmodifiers/SmoothMesh.C]: 5? Can this be an input parameter or a named constant? What does it mean?

Comment (friedmud) [reply framework/src/meshmodifiers/SmoothMesh.C]: Yep - good catch!  I accidentally left it in like that, I meant to change it to a parameter!

Comment (friedmud) [reply framework/src/meshmodifiers/SmoothMesh.C]: As far as what it means: it's how many smoothing passes.  It's essentially an iterative solver... so each iteration it moves closer to the "ideal" mesh.  After a few passes though, it's basically converged.

**Comment by friedmud** [comment]
Waiting on #12135 to be merged so I can use it here

**Comment by friedmud** [comment]
Ready to go.  Doco is complete with an animation:

smooth

**Review by permcody — APPROVED** [review]

**Pull request merged**